sdk/resource/resource_test.go

```diff
@@ -80,18 +79,30 @@ func TestMerge(t *testing.T) {
 			b:    resource.New(kv21, kv41),
 			want: []core.KeyValue{kv11, kv21, kv31, kv41},
 		},
+		{
+			name: "Merge with no overlap, no nil, not interleaved",
+			a:    resource.New(kv11, kv21),
+			b:    resource.New(kv31, kv41),
+			want: []core.KeyValue{kv11, kv21, kv31, kv41},
+		},
 		{
 			name: "Merge with common key order1",
 			a:    resource.New(kv11),
 			b:    resource.New(kv12, kv21),
-			want: []core.KeyValue{kv21, kv11},
+			want: []core.KeyValue{kv11, kv21},
 		},
 		{
 			name: "Merge with common key order2",
 			a:    resource.New(kv12, kv21),
 			b:    resource.New(kv11),
 			want: []core.KeyValue{kv12, kv21},
 		},
+		{
+			name: "Merge with common key order4",
+			a:    resource.New(kv11, kv21, kv41),
+			b:    resource.New(kv31, kv41),
+			want: []core.KeyValue{kv11, kv21, kv31, kv41},
+		},
```

Comment:
May want a couple more nil cases here

```suggestion
		},
		{
			name: "Merge with no keys",
			a:    resource.New(),
			b:    resource.New(),
			want: nil,
		},
		{
			name: "Merge with first resource no keys",
			a:    resource.New(),
			b:    resource.New(kv21),
			want: []core.KeyValue{kv21},
		},
		{
			name: "Merge with second resource no keys",
			a:    resource.New(kv11),
			b:    resource.New(),
			want: []core.KeyValue{kv11},
		},
```